### PR TITLE
feat: add validation to CRD

### PR DESCRIPTION
**Pull request by Flydiverny**

### CODE REVIEW

File: crd.yaml [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+    name: Last Sync
+    type: date
+  - JSONPath: .status.status
+    name: status
+    type: string
+  - JSONPath: .metadata.creationTimestamp
+    name: Age
+    type: date
+
+  validation:
```
Comment (silasbw) [inline crd.yaml]: do you know which version of Kubernetes introduced validation for CRDs?  I think we should use validation, but we might need to explicitly call out the minimum version required.

Comment (Flydiverny) [reply crd.yaml]: Seems to be available in 1.12 which is as far back as the official hosted kubernetes docs go now
https://v1-12.docs.kubernetes.io/docs/tasks/access-kubernetes-api/custom-resources/custom-resource-definitions/#validation 
think its a fair requirement :)

Comment (Flydiverny) [reply crd.yaml]: Heres the list I was looking for https://kubernetes.io/docs/reference/command-line-tools-reference/feature-gates/
Seems to have been enabled by default as beta in 1.9

Comment (silasbw) [reply crd.yaml]: awesome

**Pull request merged**